atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -112,6 +113,14 @@ def run(self):
             self.log.info(str(e))
             self.log.info('Canceling the compose %s', e.compose_id)
             self.odcs_client.cancel_compose(e.compose_id)
+
+            # Cancel also composes which were created by the build
+            preserve_composes = set(self.compose_ids).union(set(self.parent_compose_ids))
+            cancel_composes = list(set(self.all_compose_ids) - preserve_composes)
```

Comment:
You don't need list, you can iterate over set jut fine
```suggestion
            cancel_composes = set(self.all_compose_ids) - preserve_composes
```